recipes-extended/xen/files/libxl-retry-qmp-pci-add.patch

```diff
@@ -0,0 +1,28 @@
+--- a/tools/libxl/libxl_qmp.c
++++ b/tools/libxl/libxl_qmp.c
+@@ -849,6 +849,7 @@ int libxl__qmp_pci_add(libxl__gc *gc, in
+     libxl__json_object *args = NULL;
+     char *hostaddr = NULL;
+     int rc = 0;
++    int i = 0;
+ 
+     qmp = libxl__qmp_initialize(gc, domid);
+     if (!qmp)
+@@ -880,8 +881,17 @@ int libxl__qmp_pci_add(libxl__gc *gc, in
+     if (pcidev->permissive)
+         qmp_parameters_add_bool(gc, &args, "permissive", true);
+ 
++    
+     rc = qmp_synchronous_send(qmp, "device_add", args,
```

Comment:
IMO this call should be replaced by the loop, instead of being duplicated there.

Reply:
I'd have to initialize rc to something non-zero or else I have no way to get in the loop

Reply:
Remove this line and the `if` below, swap the sleep and the qmp call, and done :)